### DISCUSSION TITLE
MQTT function

### DISCUSSION THREAD

**Discussion by TheSimTek** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#484] [General]
Built the kit and programmed it mid 2021 for my Bestway Palm springs and haven't had any issues.  Now thinking of integrating it with Home Assistant. 

Anyone done this with the MQTT feature?  Still reading up on it and wondering if it can turn heat on/off if HA detects solar above/below certain ranges.

Thanks.

**Comment by visualapproach** [comment]
You just have to have mosquitto mqtt broker running in ha and enter the settings on the mqtt web page. Then the layzspa is auto discovered in ha. If you know how to make automations in ha, your request is basic. Trigger on solar sensor and action: turn heat on/off

**Comment by TheSimTek** [comment]
Thanks.  Will get up Home Assistant and give it a go.

**Comment by TheSimTek** [comment]
Might be missing something but this does not work.  I have both the spa and HA on the same network.  I have setup the MQTT credentials on the spa and have Mosquitto running on HA but cannot see the device.  I am running BW_2.0.0 in case there are extra steps?!?

**Reply by visualapproach** [reply]
I don't think HA autodiscovery was introduced until v3. We're on v4 now, so time to update ;-)

**Reply by TheSimTek** [reply]
Damn, I was going with the "it ain't broke so don't fix it".  I will update the software on the board in the coming weeks and give it another go.  Thanks for your excellent work.

**Comment by TheSimTek** [comment]
Now in a world of pain.  The upgrade still is not seen by HA.  The pump no longer works.  Hoping for a little guidance now to set things straight.

I built my own board two years ago for my Palm Springs.  Refer to attached images.  Perhaps the pin outs are no longer the same which may explain why it no longer works.  Last working configuration was the BW2.0.

I have tried several hardware configurations without success.  6 wire, pre 2021; 6 wire 2021 and 4 wire, 54154.  None work with PCB V1 or V2.

Once I get it working at least it will be up to date, then there is the HA issue where Mosquitto broker does not see the device.  Could it be a hardware limitation?

Hoping you can set me straight.  Thanks.  I did find this information in an excel file.  Haven't done any more until you can confirm next move.

PINOUT DSP & CIO:
--
1 - +5V
2 - GND
3 - DATA
4 - CLK
5 - SS/CS
6 - AUDIO


![Bestway_wifihack](https://user-images.githubusercontent.com/83006304/230757876-9ae9f5f8-047d-4464-85cc-95cb35e0a050.jpg)
![Bestway_wifihack_config](https://user-images.githubusercontent.com/83006304/230757878-9fd89389-479d-4a27-a8a2-99cd5522ae60.jpg)

**Reply by visualapproach** [reply]
The pinout on the schematics is the same as V1, so if you built it according to the schematics it should be V1. Did you save and restart?
Try to get it working before playing with MQTT. Disable MQTT.
Set everything from web ui and not by editing code.
I see that you built it according to an early version with resistors between the esp and the LLC. It is not recommended anymore, but I don't see why it would be a problem if it worked before. Did you run it and it worked recently? Thinking else it may be hardware issues after the winter.

**Comment by TheSimTek** [comment]
Tried version 1 board in the config but it does not work.  It was working perfectly with BW2.0 for the last two years until this afternoons upgrade.  I know the palm springs is identified as 4 wire, but mine is 6 wire.  None of the settings has worked so far.  Even tried the pinout from the excell file a few minutes ago.  Nothing.  

As suggested, MQTT disabled as I am focusing on getting this working again with the spa.

**Reply by visualapproach** [reply]
The pinout in the excel file I think is the same. Or rather, the 1=5V refers to the output pin "number", not the ESP pin. To be sure you can trace your connections back to the esp. CIO third pin (data) is connected to D7 on the ESP. Fourth (clk) to D2 and so on.

Describe what you mean by 'pump not working' as detailed as possible. Is the display lit? What happens or not when you press the display buttons. What does the display show? What about the web UI control? Sound?

**Comment by TheSimTek** [comment]
pump not working.  most configs do nothing.  one dimly lights the display but no buttons work.  another causes it to beep softly but nothing lights up.

**Reply by visualapproach** [reply]
Ok here is a step by step guide
1 Find out which PCB configuration you need (hardware config page) by following the traces on your board.
2 Find out which model you should use by checking your #defines in your previous code (2.0). Likely pre2021. Unless you have hydrojets.
3 Set the config accordingly in the hardware config page. Save and reboot.

4 After you confirmed the settings above, answer these questions
- What happens in the display when powering up the pump?
- Text in the display?
- Sounds?
- Is the ESP web server working? Post a screenshot of main page with console open (use chrome, right click and select inspect. Click tab console)
- What happens when you flip the switches on the main page?
- What happens when you press the touch buttons on the display?

**Comment by TheSimTek** [comment]
ok, will have to try this another day.  rolled back to old version and now nothing works.  suspect hardware now damaged.  Egg ok without it in circuit.  
might look at the upgraded hardware

**Comment by TheSimTek** [comment]
ok, it got the better of me.  All wiring in accordance with the following schematic is correct.

All voltages are correct.  3.3v on the board and lower side of level shifter.  4.8v on the high side.  All voltages and ground points present at the connectors.

Extremely puzzling at this point as to why it no longer works.  Will look at the code, but this is not my strong point, before upgrading it back to the latest version.

thanks again for your help.

![8ch converter](https://user-images.githubusercontent.com/83006304/230765162-448d5a44-57bc-488e-9ac1-6ad3eadf1334.png)

**Reply by TheSimTek** [reply]
note, some time back i had shorted out the resistors with a wire link.

**Comment by TheSimTek** [comment]
Ok, got my setup running again with the old BW_2.0.0 version.  

Noticed it now says "MQTT Connected" adjacent to the red temperature whereas I recall it used to say "-1".  However, HomeAssistant still does not see it.   Rebooting HA and the Spa Control says "Connect_Failed".  So its there, but nothing configurable.  More investigation required.

Now want to build another unit for the latest software version.  Please advise correct schematic to build it to.

**Reply by visualapproach** [reply]
Try a bare ESP8266. See #312

Regarding MQTT - The ESP only sends auto discovery data at startup, so if you delete the device from HA you need to restart the ESP. And v2.0.0 will not be auto discovered in HA. You need to make your own YAML sensors, but why bother since it's made for you in the later versions.

**Comment by TheSimTek** [comment]
I will build another board for backup.

This morning I upgraded again with the same settings as above config for the 6 wire, pre 2021 V1 board and it is working.  I cannot explain this.  HA is seeing everything so I can begin automation once I rectify the E02 error now present after being off for 2 days.

Thanks again for you support.

**Reply by visualapproach** [reply]
Good to hear!
